src/components/graphs/contributions/contributionsGraphContainer.js

```diff
@@ -9,34 +9,26 @@ class ContributionsGraphContainer extends React.Component {
     super(props);
     this.state = {
       chartData: [],
-      successfulFetches: 0,
-      chartType: 'default'
+      chartType: 'default',
+      loading: true
     };
   }
 
-  componentWillMount() {
+  async componentWillMount() {
     let dataArray = [],
       candidates = this.props.location.state,
       successfulFetches = this.state.successfulFetches;
 
     for (let candidate of candidates) {
       for (let campaign of candidate.data.campaigns) {
-        Client.getCampaignData(campaign.campaignId).then(data => {
-          dataArray.push(data);
-          this.setState({
-            chartData: dataArray,
-            successfulFetches: (successfulFetches += 1)
-          });
-        });
+        dataArray.push(await Client.getCampaignData(campaign.campaignId))
       }
     }
-  }
+    this.setState({
```

Comment:
since line 24 is waiting, could the lines 27 through 30 execute before the await complete? see if you find any doc around async/await and the react lifecycle